### PR TITLE
Don't check `...` in `tbl_sql()`

### LINKED ISSUE #1384: `tbl_sql()` incorrectly fails via `check_dots_used()` with duckdb
From https://github.com/tidyverse/dbplyr/pull/1222/files#r1374651419.

What are we missing?

``` r
options(conflicts.policy = list(warn = FALSE))
con <- DBI::dbConnect(duckdb::duckdb())

parquet <- tempfile(fileext = ".parquet")
arrow::write_parquet(data.frame(a = 1), parquet)

dplyr::tbl(con, parquet, check_from = FALSE, cache = TRUE)
#> Error in `tbl_sql()`:
#> ! Arguments in `...` must be used.
#> ✖ Problematic argument:
#> • cache = cache
#> ℹ Did you misspell an argument name?
#> Backtrace:
#>      ▆
#>   1. ├─dplyr::tbl(con, parquet, check_from = FALSE, cache = TRUE)
#>   2. ├─duckdb:::tbl.duckdb_connection(...)
#>   3. ├─base::NextMethod("tbl")
#>   4. └─dplyr:::tbl.DBIConnection(con, parquet, check_from = FALSE, cache = TRUE)
#>   5.   ├─dplyr::tbl(...)
#>   6.   └─dbplyr:::tbl.src_dbi(...)
#>   7.     └─dbplyr::tbl_sql(c(subclass, "dbi"), src = src, from = from, ...)
#>   8.       └─rlang (local) `<fn>`()
#>   9.         └─rlang:::check_dots(env, error, action, call)
#>  10.           └─rlang:::action_dots(...)
#>  11.             ├─base (local) try_dots(...)
#>  12.             └─rlang (local) action(...)

DBI::dbDisconnect(con, shutdown = TRUE)
```

<sup>Created on 2023-10-27 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.1 (2023-06-16)
#>  os       macOS Ventura 13.6
#>  system   aarch64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       Europe/Zurich
#>  date     2023-10-27
#>  pandoc   3.1.3 @ /opt/homebrew/bin/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version  date (UTC) lib source
#>  arrow         13.0.0.1 2023-09-22 [1] RSPM
#>  assertthat    0.2.1    2019-03-21 [1] CRAN (R 4.3.0)
#>  bit           4.0.5    2022-11-15 [1] CRAN (R 4.3.0)
#>  bit64         4.0.5    2020-08-30 [1] CRAN (R 4.3.0)
#>  cli           3.6.1    2023-03-23 [1] CRAN (R 4.3.0)
#>  DBI           1.1.3    2022-06-18 [1] CRAN (R 4.3.0)
#>  dbplyr        2.4.0    2023-10-26 [1] CRAN (R 4.3.1)
#>  digest        0.6.33   2023-07-07 [1] RSPM
#>  dplyr         1.1.3    2023-09-03 [1] CRAN (R 4.3.0)
#>  duckdb        0.9.1    2023-10-13 [1] CRAN (R 4.3.1)
#>  evaluate      0.22     2023-09-29 [1] CRAN (R 4.3.1)
#>  fansi         1.0.5    2023-10-08 [1] CRAN (R 4.3.1)
#>  fastmap       1.1.1    2023-02-24 [1] CRAN (R 4.3.0)
#>  fs            1.6.3    2023-07-20 [1] CRAN (R 4.3.0)
#>  generics      0.1.3    2022-07-05 [1] RSPM
#>  glue          1.6.2    2022-02-24 [1] CRAN (R 4.3.0)
#>  htmltools     0.5.6.1  2023-10-06 [1] CRAN (R 4.3.1)
#>  knitr         1.44     2023-09-11 [1] CRAN (R 4.3.0)
#>  lifecycle     1.0.3    2022-10-07 [1] RSPM
#>  magrittr      2.0.3    2022-03-30 [1] CRAN (R 4.3.0)
#>  pillar        1.9.0    2023-03-22 [1] CRAN (R 4.3.0)
#>  pkgconfig     2.0.3    2019-09-22 [1] CRAN (R 4.3.0)
#>  purrr         1.0.2    2023-08-10 [1] CRAN (R 4.3.0)
#>  R.cache       0.16.0   2022-07-21 [1] CRAN (R 4.3.0)
#>  R.methodsS3   1.8.2    2022-06-13 [1] RSPM
#>  R.oo          1.25.0   2022-06-12 [1] CRAN (R 4.3.0)
#>  R.utils       2.12.2   2022-11-11 [1] CRAN (R 4.3.0)
#>  R6            2.5.1    2021-08-19 [1] RSPM
#>  reprex        2.0.2    2022-08-17 [1] RSPM
#>  rlang         1.1.1    2023-04-28 [1] CRAN (R 4.3.0)
#>  rmarkdown     2.25     2023-09-18 [1] RSPM
#>  rstudioapi    0.15.0   2023-07-07 [1] CRAN (R 4.3.0)
#>  sessioninfo   1.2.2    2021-12-06 [1] CRAN (R 4.3.0)
#>  styler        1.10.2   2023-08-29 [1] CRAN (R 4.3.0)
#>  tibble        3.2.1    2023-03-20 [1] CRAN (R 4.3.0)
#>  tidyselect    1.2.0    2022-10-10 [1] RSPM
#>  utf8          1.2.4    2023-10-22 [1] CRAN (R 4.3.1)
#>  vctrs         0.6.4    2023-10-12 [1] CRAN (R 4.3.1)
#>  withr         2.5.1    2023-09-26 [1] CRAN (R 4.3.1)
#>  xfun          0.40     2023-08-09 [1] CRAN (R 4.3.0)
#>  yaml          2.3.7    2023-01-23 [1] CRAN (R 4.3.0)
#> 
#>  [1] /Users/kirill/Library/R/arm64/4.3/library
#>  [2] /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1384

### CODE REVIEW

**Review by krlmlr — COMMENTED** [review]
Should we add a test here, derived from https://github.com/tidyverse/dbplyr/issues/1384#issuecomment-1791533590, which, I believe, is an accurate replica of the original problem?

**Comment by hadley** [comment]
I doubt it's worth it.

File: R/tbl-sql.R [hunk 1]
```diff
@@ -14,7 +14,6 @@
 #'   multiple `tbl` objects.
 #' @param check_from Check if `from` is likely misspecified SQL or a table in a schema.
 tbl_sql <- function(subclass, src, from, ..., vars = NULL, check_from = TRUE) {
```
Comment (krlmlr) [inline R/tbl-sql.R:16]:
```suggestion
tbl_sql <- function(subclass, src, from, ..., vars = NULL, check_from = TRUE) {
  # Can't use check_dots_used(), #1429
```

**Review by krlmlr — APPROVED** [review]
In lieu of a test?

**Review by mgirlich — APPROVED** [review]

**Pull request merged**